sopel/modules/reddit.py

```diff
@@ -258,3 +302,38 @@ def get_channel_spoiler_free(bot, trigger):
         bot.say('%s is flagged as spoiler-free' % channel)
     else:
         bot.say('%s is flagged as spoilers-allowed' % channel)
+
+
+@rule(r"^(r|u)\/([^\s/]+)")
```

Comment:
According to this [Reddit help post](https://www.reddit.com/r/help/comments/1ttv80/what_are_the_valid_usernamecharacters/) the valid characters are `a-zA-Z-_` and nothing else.

Here is my take on that:

``` python
>>> import re
>>> regex = re.compile(r'^(?P<prefix>r|u)\/(?P<id>[a-zA-Z-_]+)')
>>> regex.match('u/toto')
<_sre.SRE_Match object; span=(0, 6), match='u/toto'>
>>> regex.match('u/toto').group(0)
'u/toto'
>>> regex.match('u/toto').group(1)
'u'
>>> regex.match('u/toto').group(2)
'toto'
>>> regex.match('u/toto').group('prefix')
'u'
>>> regex.match('u/toto').group('id')
'toto'
```